src/index.js

```diff
@@ -1,15 +1,44 @@
 #!/usr/bin/env node
+// @ts-check
 
-import { Package, Step } from "./declarations/types";
-import { execSync } from "child_process";
-import fs from "fs-extra";
-import path from "path";
-import prompts from "prompts";
-import validateName from "validate-npm-package-name";
+const { execSync } = require("child_process");
+const fs = require("fs-extra");
+const path = require("path");
+const prompts = require("prompts");
+const validateName = require("validate-npm-package-name");
 
-const getApplicationId = (token: string): string | null => {
+/**
+ * @typedef {() => void} StepAction
+ */
+
+/**
+ * @typedef {Object} Package
+ * @property {string} name
+ * @property {string} version
+ * @property {string} description
+ * @property {Record<string, string> | undefined | null} scripts
+ * @property {Record<string, string> | undefined | null} dependencies
+ * @property {Record<string, string> | undefined | null} devDependencies
```

Comment:
was testing this typedef like so:
```
/** @type {Package} */
const temp = {
  name: "asdf",
  version: "asdf",
  description: "asdf",
};
```
vscode was telling me scripts/dep/devdeps were missing from this declaration.

it appears the correct way to define optional properties is to wrap the field name with `[]`, so this should suffice:
```
/**
 * @typedef {Object} Package
 * @property {string} name
 * @property {string} version
 * @property {string} description
 * @property {Record<string, string>} [scripts]
 * @property {Record<string, string>} [dependencies]
 * @property {Record<string, string>} [devDependencies]
 */
```
https://jsdoc.app/tags-param.html#optional-parameters-and-default-values